src/middleware/cloudWatch.ts

```diff
@@ -37,44 +34,71 @@ export class CloudWatchMiddleware
       .body as WrappedDocument<HealthCertDocument>;
     const data = getData(wrappedDocument);
     const provider = data.issuers[0].identityProof?.location ?? "UNKNOWN";
-    this.provider = provider;
-    const subDomain = this.extractSubDomain(provider);
-    trace(`provider ${provider} attempting to notarise pdt...`);
-    trace(`subDomain ${subDomain} attempting to notarise pdt...`);
+    this.specificDomain = provider;
+    const aggregateDomain = this.toAggregateDomain(this.specificDomain);
+    trace(
+      `specificDomain ${this.specificDomain} attempting to notarise pdt...`
+    );
+    trace(`aggregateDomain ${aggregateDomain} attempting to notarise pdt...`);
   };
 
   after = async (req: Request): Promise<void> => {
-    const { body } = req.response;
+    const { body, statusCode } = req.response;
+    if (statusCode !== 200) {
+      logError(
+        "error encountered, logging for successful notarisation skipped"
+      );
+      return;
+    }
+
     try {
       const notarisationResult: NotarisationResult = JSON.parse(body);
       const { notarisedDocument } = notarisationResult;
       const data: HealthCertDocument | PDTHealthCertV2Document =
         getData(notarisedDocument);
-      let testName = "";
-      if (data?.version === "pdt-healthcert-v2.0") {
-        // @ts-ignore
-        const observationResource = data.fhirBundle.entry?.find(
-          (entr: any) => entr.resource.resourceType === "Observation"
-        )?.resource;
-        const observation = parsers(observationResource) as ObservationV2;
-        testName = observation.testType?.display || "";
+      let testTypes: string[] = [];
+      const validTestTypes = ["art", "pcr", "ser"];
+      if (data.version === "pdt-healthcert-v2.0") {
+        // version 2
+        if (typeof data.type === "string") {
+          testTypes = [data.type];
+        } else if (Array.isArray(data.type)) {
+          testTypes = data.type;
+        }
+        testTypes.forEach((test) => test.toLowerCase());
+        const allValid: boolean = testTypes.every((test) =>
+          validTestTypes.includes(test)
```

Comment:
`test` can be lower or upper case, so it will be good use like this `test.toLocaleLowerCase()`. Else i'm getting `ART are not valid` error when try with valid `ART` type.

Reply:
line 68 already has `testTypes.forEach((test) => test.toLowerCase());`?

Reply:
ah.. yea, there is no error log now. i think i test without pull new changes. nvm , it's look good. you can merge.